### PR TITLE
EPMDJ-8045 Fixed: coverage for the test is overwritten, not merged

### PR DESCRIPTION
**Pull request by ViktoriiaLebedeva**

### CODE REVIEW

File: admin-part/src/main/kotlin/Coverage.kt [hunk 1]
```diff
@@ -165,27 +165,25 @@ internal fun Map<String, BundleCounter>.coveragesByTestType(
...
-                    session.asSequence()
-                        .groupBy { TypedTest(it.testName, testType) }
-                        .mapValuesTo(testsWithEmptyBundle) {
-                            it.value.asSequence().bundle(context, classBytes)
-                        }
-                }
+    cache: Map<TypedTest, BundleCounter>,
+): Map<TypedTest, BundleCounter> = run {
+    val testsWithEmptyBundle = values.asSequence().flatten().testsWithBundle()
+    testsWithEmptyBundle.putAll(cache)
```
Comment (LikhanovMaxim) [inline admin-part/src/main/kotlin/Coverage.kt:171]: testsWithEmptyBundle

File: admin-part/src/main/kotlin/Scope.kt [hunk 2]
```diff
@@ -74,7 +81,8 @@ class ActiveScope(
         )
     )
 
-    private val _handler = atomic<ActiveScopeHandler?>(null)
+    private val _coverageHandler = atomic<CoverageHandler?>(null)
```
Comment (LikhanovMaxim) [inline admin-part/src/main/kotlin/Scope.kt:84]: Realtime

File: admin-part/src/main/kotlin/Scope.kt [hunk 3]
```diff
@@ -97,10 +105,34 @@ class ActiveScope(
...
+                }.takeIf { it.isNotEmpty() }?.reduce { m1, m2 ->
+                    m1.apply { putAll(m2) }
+                } ?: emptyMap()
+                it(probesByTests)
+                delay(500)
+            }
+        }
+    }
+
+    fun initScopeHandler(handler: CoverageHandler): Boolean = _coverageHandler.getAndUpdate {
```
Comment (LikhanovMaxim) [inline admin-part/src/main/kotlin/Scope.kt:129]: naming

File: admin-part/src/main/kotlin/Scope.kt [hunk 4]
```diff
@@ -97,10 +105,34 @@ class ActiveScope(
         }
     }
 
-    fun initScopeHandler(handler: ActiveScopeHandler): Boolean = _handler.getAndUpdate {
+    private val activeSessionJob = AsyncJobDispatcher.launch {
```
Comment (LikhanovMaxim) [inline admin-part/src/main/kotlin/Scope.kt:108]: naming

File: admin-part/src/main/kotlin/Scope.kt [hunk 5]
```diff
@@ -97,10 +105,34 @@ class ActiveScope(
...
+        while (true) {
+            delay(250)
```
[the remark this answers is not in the document]
Comment (ViktoriiaLebedeva) [reply admin-part/src/main/kotlin/Scope.kt]: I set it to 3 seconds because this is how often the autotest agent sends the tests

**Pull request merged**